### DISCUSSION TITLE
Sionna RT Coverage map

### DISCUSSION THREAD

**Discussion by achaitu** [NVlabs/sionna#225] [Q&A]
Hi, I'm new to EM and raytracing. I want to generate the signal strength at every cell in a scene of size `M*M`. I do not need pixel-by-pixel granularity, so I am using coverage_map with cm_cell_size=(4.0, 4.0) to effectively make my scene `M/4*M/4`. So, I want to get the signal strength at each of the cells in the `M/4*M/4` grid. If I compute using coverage_map, I can only see a few non-zero elements in the coverage_map (by accessing the coverage_map.as_tensor), see the fig below, (even the fig uploaded in https://nvlabs.github.io/sionna/api/rt.html#coverage-maps) has many 0's I presume.

<img width="565" alt="image" src="https://github.com/NVlabs/sionna/assets/28532850/476f9d3f-a114-43cb-9cbd-0db329e66d8c">

So, does that mean the signal power at those cells is 0 or not enough rays have reached these cells effectively giving the power a 0? Is there anything else I can do to get the power at any required cell by calling any other functions or re-running the coverage_map multiple times?

Also, if I want to generate the signal strengths in 2D, (say even my scene is in 2D or I sliced a 3D scene) is this possible with Sionna RT? Because I am giving the z-coordinate to be zero for the Transmitter, and am observing the coverage map (by setting the cm_center=(M/2, M/2, 0) instead of None that defaults to (M/2, M/2, 1.5)) to be all 0s. Why is this the case? 

My end goal: Given a single transmitter, I want to know the signal power at every cell (of `4*4`) in an `M*M` scene where the transmitter and receivers are all on the same plane (effectively considering a 3D scene and cutting only a plane parallel to XY plane and putting the transmitter on that plane and calculating the signal strength of every grid cell on that plane)

**Comment by jhoydis** [comment]
Hi,

There are multiple things you can do to get a better precision of a coverage map:
- Simply shoot more rays using the ``num_samples`` parameters
- Increase the ``max_depth``
- Activate diffraction and maybe scattering (if your scene is configured for this)

The specific problem you observe is very likely related to the fact that the transmitter is located exactly within the plane that defines the coverage map. The line-of-sight paths will essentially  never intersect with this plane as they lie exactly in it (up to some numerical imprecisions). You could try to add a very small hight difference, for example z=1e-3, and see if this solves the problem.
